### ISSUE TITLE
ERROR: Failed to connect to "/var/run/keyd.socket",

### ISSUE THREAD

**Issue by rafaeluriarte** [rvaiya/keyd#743]
Hi, 

I am having troubles with keyd-application-mapper. I am using debian and X. I added my user to the keyd group, the daemon is running, but I keep getting the following error when I change window (and it does not work) - keyd v2.4.3 (02c77af):

ERROR: Failed to connect to "/var/run/keyd.socket", make sure the daemon is running and you have permission to access the socket.
Active window: guake|guake
ERROR: Failed to connect to "/var/run/keyd.socket", make sure the daemon is running and you have permission to access the socket.

The socket is also there:
srw-rw---- 1 root keyd 0 May  9 11:46 /var/run/keyd.socket

Any idea on how to solve it?

Thanks!!

**Comment by rafaeluriarte** [comment]
solved adding a chown on the file on the poststart in the /system/multi-user.target.wants/keyd.service

**Comment by ahmet-umut** [comment]
> solved adding a chown on the file on the poststart in the /system/multi-user.target.wants/keyd.service

Can you explain? I started to have the same problem and when I try to change the owner of the file thou mentionedst, it says that the file is not found.

**Comment by rafaeluriarte** [comment]
In my case, the issue was that the socket was being created with root permissions. When I tried to access it with my user account to change the context according to the application, it couldn't read the socket file.

To resolve this, I changed the file permissions using a post-start script. After the socket file is created, the script changes the permissions to allow my user account to access the socket file.
